client/src/shared/planCard/index.jsx

```diff
@@ -7,53 +7,340 @@ import { Grid } from '@material-ui/core';
 import { useTheme } from '@material-ui/core/styles';
 import UnsafeRender from 'components/UnsafeRender';
 import { Text } from 'shared';
+import moment from 'moment'
+import styled from 'styled-components';
+import BookmarkOutlinedIcon from '../../assets/bookmark-outlined-icon.svg';
+import BookmarkFilledIcon from '../../assets/bookmark-filled-icon.svg';
+import DropPinIcon from '../../assets/drop-pin-icon.svg';
+import { openModal } from '../../redux/modal/slice';
+import { isFavoritePlan } from '../../pages/Plan/hooks';
+import { UserSelectors } from '../../redux/selectors';
+import { useDispatch } from 'react-redux';
+import { subscribeUserToPlan, unsubscribeUserToPlan } from '../../pages/Plan/controller';
+
 
 const PlanCard = ({ plan }) => {
 	const theme = useTheme();
+	const tagsWrapperRef = useRef(null);
+	const [tags, setTags] = useState(plan?.tags || []);
+    const [ isFavPlan, setIsFavPlan ] = useState(false);
+	const tagsRef = useRef([...(plan?.tags || []).map(() => createRef()), createRef()]);
+    const dispatch = useDispatch();
+    const { isAuthenticated, user } = UserSelectors();
+
+    useEffect(() => {
+        const handler = async () => {
+            await getIsFav();
+        };
+        handler();
+    }, [getIsFav]);
```

Comment:
Also, can we please extract the SubscriptionHandler to another FC? I mean, it will represent the subscription status for a plan per person, and will render its status and handle like/unlike.

In this case, we can share code between plan page and plan card
Something like that:

```
export const PlanSubscriptionStatus = ({ planId }) => {
// Getting 'liked' status from store/ server etc
const isSubscribed = useSelector(.....);
// here rendering the bookmark- full or not
return <bookmarkIcon filled={isSubscribed} tooltipText={isSubscribed:'unsubscribe': "subscribe"})
}
```